llvm/lib/Target/SPIRV/SPIRVModuleAnalysis.cpp

```diff
@@ -1218,6 +1223,17 @@ void addInstrRequirements(const MachineInstr &MI,
       Reqs.addCapability(SPIRV::Capability::SplitBarrierINTEL);
     }
     break;
+  case SPIRV::OpSDot:
+  case SPIRV::OpUDot: {
+    if (ST.canUseExtension(SPIRV::Extension::SPV_KHR_integer_dot_product)) {
+      Reqs.addExtension(SPIRV::Extension::SPV_KHR_integer_dot_product);
+    }
+    Reqs.addCapability(SPIRV::Capability::DotProductKHR);
+    Reqs.addCapability(SPIRV::Capability::DotProductInputAllKHR);
+    Reqs.addCapability(SPIRV::Capability::DotProductInput4x8BitKHR);
+    Reqs.addCapability(SPIRV::Capability::DotProductInput4x8BitPackedKHR);
```

Comment:
You don't need all of the requirements. This can be determined based on the type of the operands:

> Vector 1 and Vector 2 must be either 32-bit integers (enabled by DotProductInput4x8BitPackedKHR) or vectors of integer type (enabled by DotProductInput4x8BitKHR or DotProductInputAllKHR).